### PR TITLE
feat: compute and verify hash of synced ContractClass

### PR DESCRIPTION
**Pull request by tdelabro**

### CODE REVIEW

File: crates/client/sync/src/utils/convert.rs [hunk 1]
```diff
@@ -234,12 +234,12 @@ pub fn convert_and_verify_class(
             let DbClassUpdate { class_hash, contract_class, compiled_class_hash } = class_update;
 
             // TODO(class_hash): uncomment this when the class hashes are computed correctly accross the entire state
-            // let expected =
-            //     contract_class.class_hash().map_err(|e| ConvertClassError::ComputeClassHashError(e.to_string()))?;
-            // if class_hash != expected {
-            //     log::warn!("Mismatched class hash: 0x{:x}", class_update.class_hash);
-            //     // return Err(ConvertClassError::MismatchedClassHash { expected, got: class_hash });
-            // }
+            let expected =
```
Comment (apoorvsadana) [inline crates/client/sync/src/utils/convert.rs]:
```suggestion
            let expected_class_hash =
```

File: crates/client/sync/src/utils/convert.rs [hunk 2]
```diff
@@ -234,12 +234,12 @@ pub fn convert_and_verify_class(
...
+                contract_class.class_hash().map_err(|e| ConvertClassError::ComputeClassHashError(e.to_string()))?;
+            if class_hash != expected {
+                log::warn!("Mismatched class hash: 0x{:x}", class_update.class_hash);
+                return Err(ConvertClassError::MismatchedClassHash { expected, got: class_hash });
+            }
 
             let compiled_class =
```
Comment (apoorvsadana) [inline crates/client/sync/src/utils/convert.rs:223]: maybe not the scope of this PR but why don't check check compiled hashes as well?

**Comment by antiyro** [comment]
I've tested your branch against mainnet and it fails on block #5076 for class hash: `0x7c36dd12364567ab91f35f4d64920a3555735e3a6916ced2236958a257d30db` with the following error:

```sh
[2024-07-24 12:36:34 INFO] ✨ Imported #5076 (0x94b1c2...970cf4) and updated state root (0x19a4b1...66a93f)
Error: Converting classes

Caused by:
    Mismatched class hash, expected 0x7c36dd12364567ab91f35f4d64920a3555735e3a6916ced2236958a257d30db; got 0x31a585ff31b2afa2274482c7b67e8780b4d8041417fb58698fb3280d6b85f9d
[2024-07-24 12:36:36 INFO] ⏳ Gracefully closing the database...
```

**Comment by antiyro** [comment]
You can use `--starting-block <block_number>` to avoid having to sync from block 0

**Comment by antiyro** [comment]
closing this as it is solved by #268

**Pull request closed without merging**